File: infra/src/main/java/com/example/jpa/entity/movie/Movie.java
package com.example.jpa.entity.movie;

import com.example.jpa.entity.BaseEntity;
import jakarta.persistence.*;

import java.time.LocalDate;

@Entity
public class Movie extends BaseEntity {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private String thumbnail;

    @Enumerated(EnumType.STRING)
    private Genre genre;

    @Enumerated(EnumType.STRING)
    private Grade grade;

    private LocalDate releaseDate;

    private Long runningTime;

}

File: movies/src/main/kotlin/com/redis/movies/domain/Movie.kt
package com.redis.movies.domain

import jakarta.persistence.*
import java.time.LocalDateTime

@Table(name = "movie")
@Entity
class Movie(
    @Column(name = "title")
    val title: String,

    @Column(name = "film_ratings")
    @Enumerated(EnumType.STRING)
    val filmRatings: String,

    @Column(name = "release_date")
    val releaseDate: LocalDateTime,

    @Column(name = "thumbnail_image_path")
    val thumbnailImagePath: String,

    @Column(name = "running_time")
    val runningTime: Long,

    @OneToOne(mappedBy = "movie", cascade = [CascadeType.ALL], orphanRemoval = true)
    val genre: MovieGenre,

    @Column(name = "theater_name")
    val theaterName: String,

    val screeningSchedules: ScreeningSchedules = ScreeningSchedules(),

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null
) {

}